client/my-sites/themes/themes-selection.jsx

```diff
@@ -180,7 +180,7 @@ const ConnectedThemesSelection = connect(
 			page,
 			tier: config.isEnabled( 'upgrades/premium-themes' ) ? tier : 'free',
 			filter: compact( [ filter, vertical ] ).join( ',' ),
-			number: 20
+			number: isJetpack && ! ( queryWpcom === true ) ? 2000 : 20
```

Comment:
Can we add an inline comment to remind us that this is a hack, and that a more proper solution would be to add pagination (and search, see https://github.com/Automattic/jetpack/pull/2562 for reference) to the Jetpack endpoint?